aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -1719,14 +1719,14 @@ void do_avg_pool_nhwc_on_AVX_n(
       // first quantize using AVX2 or AVX512 using 32 lanes, then 8, finally falls
       // back to single
 #ifdef CPU_CAPABILITY_AVX2
-      QuantizeAvx2<T>(
+      QuantizeAvx2<typename T::underlying>(
```

Comment:
May I know why this change is necessary?

Reply:
The initial reason for doing this is I failed to deduce `c10:qint8` or `c10:quint8` while invoking `QuantizeAvx2/QuantizeAvx512` in the test. And I found it is non-necessary to know about `c10:qint8` or `c10:quint8` in `QuantizeAvx2/QuantizeAvx512`. What need to know is only the `underlying`. I also found `RequantizeAvx512/RequantizeAvx2` is using the `underlying` directly. So I do this change.